### PR TITLE
feat(APIv2): add a DELETE and POST endpoints for host severities

### PR DESCRIPTION
**Pull request by abuathier**
## Description

Jira: MON-15458

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [X] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [X] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Look at the Jira ticket.

#### Community contributors & Centreon team

- [X] I have followed the **coding style guidelines** provided by Centreon
- [X] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [X] I have commented my code, especially **hard-to-understand areas** of the PR.
- [X] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 1]
```diff
@@ -17298,3 +17298,24 @@ msgstr "Vous n'êtes pas autorisé à accéder aux sévérités d'hôtes"
 
 msgid "Error while searching for host severities"
 msgstr "Erreur durant la recherche de sévérités d'hôtes"
+
+msgid "You are not allowed to delete host severities"
+msgstr "Vous n'êtes pas autorisé à créer des sévérités d'hôtes"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17303]:
```suggestion
msgstr "Vous n'êtes pas autorisé à supprimer des criticités d'hôtes"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 2]
```diff
@@ -17298,3 +17298,24 @@ msgstr "Vous n'êtes pas autorisé à accéder aux sévérités d'hôtes"
...
+
+msgid "Error while deleting host severity"
+msgstr "Erreur durant la suppression d'une sévérité d'hôtes"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17306]:
```suggestion
msgstr "Erreur durant la suppression d'une criticité d'hôtes"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -17298,3 +17298,24 @@ msgstr "Vous n'êtes pas autorisé à accéder aux sévérités d'hôtes"
...
+
+msgid "Error while creating host severity"
+msgstr "Erreur lors de la création d'une sévérité d'hôtes"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17309]:
```suggestion
msgstr "Erreur lors de la création d'une criticité d'hôtes"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -17298,3 +17298,24 @@ msgstr "Vous n'êtes pas autorisé à accéder aux sévérités d'hôtes"
...
+
+msgid "You are not allowed to create host severities"
+msgstr "Vous n'êtes pas autorisé à créer des sévérités d'hôtes"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17312]:
```suggestion
msgstr "Vous n'êtes pas autorisé à créer des criticités d'hôtes"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -17298,3 +17298,24 @@ msgstr "Vous n'êtes pas autorisé à accéder aux sévérités d'hôtes"
...
+
+msgid "Error while retrieving recently created host severity"
+msgstr "Error lors de la récupération de la sévérité d'hôtes nouvellement créée"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17315]:
```suggestion
msgstr "Error lors de la récupération de la criticité d'hôtes nouvellement créée"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -17298,3 +17298,24 @@ msgstr "Vous n'êtes pas autorisé à accéder aux sévérités d'hôtes"
...
+
+msgid "Host severity name already exists"
+msgstr "Ce nom de sévérité d'hôtes existe déjà"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17318]:
```suggestion
msgstr "Ce nom de criticité d'hôtes existe déjà"
```

File: centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -17298,3 +17298,24 @@ msgstr "Vous n'êtes pas autorisé à accéder aux sévérités d'hôtes"
...
+
+msgid "The host severity icon with id '%d' does not exist"
+msgstr "L'icône de sévérité d'hôtes avec l'identifiant '%d' n'existe pas"
```
Comment (cg-tw) [inline centreon/lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:17321]:
```suggestion
msgstr "L'icône de criticité d'hôtes avec l'identifiant '%d' n'existe pas"
```

File: centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+                );
+                $presenter->setResponseStatus(
+                    new ConflictResponse(HostSeverityException::hostNameAlreadyExists())
+                );
+            } elseif (
+                0 === $request->iconId
+                || ! $this->readViewImgRepository->existsOne($request->iconId)
+            ) {
+                $this->error(
+                    'Host severity icon does not exists',
```
Comment (cg-tw) [inline centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php:84]:
```suggestion
                    'Host severity icon does not exist',
```

File: centreon/tests/php/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverityTest.php [hunk 9]
```diff
@@ -0,0 +1,238 @@
...
+
+    ($this->useCase)($this->request, $this->presenter);
+
+    expect($this->presenter->getResponseStatus())
+        ->toBeInstanceOf(ForbiddenResponse::class)
+        ->and($this->presenter->getResponseStatus()->getMessage())
+        ->toBe(HostSeverityException::addNotAllowed()->getMessage());
+});
+
+it('should present an ConflictResponse when name is already used', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverityTest.php:103]:
```suggestion
it('should present a ConflictResponse when name is already used', function (): void {
```

File: centreon/tests/php/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverityTest.php [hunk 10]
```diff
@@ -0,0 +1,238 @@
...
+    );
+    ($this->useCase)($this->request, $this->presenter);
+
+    expect($this->presenter->getResponseStatus())
+        ->toBeInstanceOf(InvalidArgumentResponse::class)
+        ->and($this->presenter->getResponseStatus()?->getMessage())
+        ->toBe($expectedException->getMessage());
+});
+
+it('should throw an ConflictResponse if the host severity icon does not exist', function (): void {
```
Comment (cg-tw) [inline centreon/tests/php/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverityTest.php:149]:
```suggestion
it('should throw a ConflictResponse if the host severity icon does not exist', function (): void {
```

**Review by cg-tw — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

File: centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php [hunk 11]
```diff
@@ -0,0 +1,145 @@
...
+                );
+            } else {
+                $newHostSeverity = new NewHostSeverity(
+                    trim($request->name),
+                    trim($request->alias),
+                    $request->level,
+                    $request->iconId,
+                );
+                $newHostSeverity->setActivated($request->isActivated);
+                $newHostSeverity->setComment($request->comment ? trim($request->comment) : null);
```
Comment (callapa) [inline centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php:98]:
```suggestion
                $newHostSeverity->setComment($request->comment ?$request->comment : null);
```
Same remark as above

File: centreon/src/Core/HostSeverity/Application/UseCase/DeleteHostSeverity/DeleteHostSeverity.php [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+                        ['hostseverity_id' => $hostSeverityId]
+                    );
+                    $presenter->setResponseStatus(new NotFoundResponse('Host severity'));
+                }
+            } elseif ($this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ_WRITE)) {
+                $accessGroups = $this->readAccessGroupRepositoryInterface->findByContact($this->user);
+
+                if ($this->readHostSeverityRepository->existsByAccessGroups($hostSeverityId, $accessGroups)) {
+                    $this->writeHostSeverityRepository->deleteById($hostSeverityId);
+                    $presenter->setResponseStatus(new NoContentResponse());
```
Comment (callapa) [inline centreon/src/Core/HostSeverity/Application/UseCase/DeleteHostSeverity/DeleteHostSeverity.php:74]: Log missing

**Review by callapa — CHANGES_REQUESTED** [review]

File: centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+
+                $hostSeverityId = $this->writeHostSeverityRepository->add($newHostSeverity);
+                $hostSeverity = $this->readHostSeverityRepository->findById($hostSeverityId);
```
Comment (callapa) [inline centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php:101]: missing log

File: centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php (hunk 11 above)
Comment (abuathier) [reply centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php]: ✅ done with added tests

File: centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+                    'Host severity icon does not exist',
+                    ['hostseverity_name' => trim($request->name)]
+                );
+                $presenter->setResponseStatus(
+                    new ConflictResponse(HostSeverityException::iconDoesNotExist($request->iconId))
+                );
+            } else {
+                $newHostSeverity = new NewHostSeverity(
+                    trim($request->name),
+                    trim($request->alias),
```
[the remark this answers is not in the document]
Comment (abuathier) [reply centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php]: ✅ done with added tests

File: centreon/src/Core/HostSeverity/Application/UseCase/DeleteHostSeverity/DeleteHostSeverity.php (hunk 12 above)
Comment (abuathier) [reply centreon/src/Core/HostSeverity/Application/UseCase/DeleteHostSeverity/DeleteHostSeverity.php:74]: ✅ done

File: centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php (hunk 13 above)
Comment (abuathier) [reply centreon/src/Core/HostSeverity/Application/UseCase/AddHostSeverity/AddHostSeverity.php:101]: ✅ done

**Review by callapa — APPROVED** [review]

**Pull request merged**